### DISCUSSION TITLE
Xaml PropertyInfo does not update state

### DISCUSSION THREAD

**Discussion by EricNgo1972** [MarimerLLC/csla#3638] [Questions]
**Describe the bug**

In a business object, create two properties: 

1. **PasswordProperty** and 
2. **RetypePasswordProperty**.  

Implement a business rule to validate if the user entered the same password in both fields. Set an error message for the **RetypePasswordProperty** in case the validation fails. 

Note that when the user enters a value for **PasswordProperty**, the Xaml PropertyInfo bound to **RetypePasswordProperty** does not update its state. The state only updates when the user enters a value for **RetypePasswordProperty**.

```c#
  private class MatchingNewPasswords : BusinessRule
        {
            public MatchingNewPasswords(IPropertyInfo p1, IPropertyInfo p2) : base()
            {
                PrimaryProperty = p2;
                InputProperties = new List<Csla.Core.IPropertyInfo> { p1, p2 };
            }
            protected override void Execute(IRuleContext context)
            {
                string pass1 = context.GetInputValue<string>(InputProperties[0]);
                string pass2 = context.GetInputValue<string>(InputProperties[1]);
                if (pass1 != pass2)
                    context.AddErrorResult("The re-typed password does not match the newly entered password");
            }

        }
```

**Version and Platform**
CSLA version: 6.2.2
Platform:  Xamarin

**Comment by EricNgo1972** [comment]
Currently, I work around by listening the ValidationComplete in ContentPage and call UpdateState for each XAML PropertyInfo in the page. 

Update:  The issue can be resolved by using an asynchronous rule instead. Asynchronous rules change the busy status of input properties, which in turn updates the state of the PropertyInfo.

**Comment by rockfordlhotka** [comment]
Do you have a dependency rule set up between the two properties?

**Comment by EricNgo1972** [comment]
Yes, the dependency is setup correctly. The rule is checked when any of the properties change, but the checking result is not surfaced to the PropertyInfo View. 
However, the same rule derived from asynchronous methods works fine."

**Comment by rockfordlhotka** [comment]
This sounds like a difference in how `PropertyChanged` is raised between sync and async rules. That's possible - I'm pretty sure sync rules minimize raising the event to avoid spamming the UI, while async rules _must_ raise the event because we don't know when the async rule will complete.
